scylla/tests/integration/tablets.rs

```diff
@@ -0,0 +1,601 @@
+    target: (Arc<Node>, Option<u32>),
+}
+
+impl LoadBalancingPolicy for SingleTargetLBP {
+    fn pick<'a>(
+        &'a self,
+        _query: &'a RoutingInfo,
+        _cluster: &'a ClusterData,
+    ) -> Option<(NodeRef<'a>, Option<u32>)> {
+        Some((&self.target.0, self.target.1))
+    }
+
+    fn fallback<'a>(
+        &'a self,
+        _query: &'a RoutingInfo,
+        _cluster: &'a ClusterData,
+    ) -> FallbackPlan<'a> {
+        Box::new(std::iter::empty())
+    }
+
+    fn name(&self) -> String {
+        "SingleTargetLBP".to_owned()
+    }
+}
+
+async fn send_statement_everywhere(
+    session: &Session,
+    cluster: &ClusterData,
+    statement: &PreparedStatement,
+    values: &dyn SerializeRow,
+) -> Result<Vec<QueryResult>, QueryError> {
+    let tasks = cluster.get_nodes_info().iter().flat_map(|node| {
+        let shard_count: u16 = node.sharder().unwrap().nr_shards.into();
+        (0..shard_count).map(|shard| {
+            let mut stmt = statement.clone();
+            let values_ref = &values;
+            let policy = SingleTargetLBP {
+                target: (node.clone(), Some(shard as u32)),
+            };
+            let execution_profile = ExecutionProfile::builder()
+                .load_balancing_policy(Arc::new(policy))
+                .build();
+            stmt.set_execution_profile_handle(Some(execution_profile.into_handle()));
+
+            async move { session.execute(&stmt, values_ref).await }
+        })
+    });
+
+    try_join_all(tasks).await
+}
+
+async fn send_unprepared_query_everywhere(
+    session: &Session,
+    cluster: &ClusterData,
+    query: &Query,
+) -> Result<Vec<QueryResult>, QueryError> {
+    let tasks = cluster.get_nodes_info().iter().flat_map(|node| {
+        let shard_count: u16 = node.sharder().unwrap().nr_shards.into();
+        (0..shard_count).map(|shard| {
+            let mut stmt = query.clone();
+            let policy = SingleTargetLBP {
+                target: (node.clone(), Some(shard as u32)),
+            };
+            let execution_profile = ExecutionProfile::builder()
+                .load_balancing_policy(Arc::new(policy))
+                .build();
+            stmt.set_execution_profile_handle(Some(execution_profile.into_handle()));
+
+            async move { session.query(stmt, &()).await }
+        })
+    });
+
+    try_join_all(tasks).await
+}
+
+fn frame_has_tablet_feedback(frame: ResponseFrame) -> bool {
+    let response =
+        scylla_cql::frame::parse_response_body_extensions(frame.params.flags, None, frame.body)
+            .unwrap();
+    match response.custom_payload {
+        Some(map) => map.contains_key("tablets-routing-v1"),
+        None => false,
+    }
+}
+
+fn count_tablet_feedbacks(
+    rx: &mut mpsc::UnboundedReceiver<(ResponseFrame, Option<TargetShard>)>,
+) -> usize {
+    std::iter::from_fn(|| rx.try_recv().ok())
+        .map(|(frame, _shard)| frame_has_tablet_feedback(frame))
+        .filter(|b| *b)
```

Comment:
nit: why not `filter_map` ?

Reply:
`filter_map` expects `Option`, not `bool`, this would require code like `.filter_map(|(frame, _shard)| frame_has_tablet_feedback(frame).then_some(()))`. I really don't like this `.then_some(())` call on `bool`. Alternative is explicit match, but that's a lot of code.
map + filter still seems to me like the easiest option here.